sustainability/src/main/webapp/WEB-INF/views/search/results.jsp

```diff
@@ -13,9 +13,48 @@
 	</small>
 </h1>
 
-
 <ul class="list-group">
-	<c:forEach items="${results}" var="entry">
-		<li class="list-group-item"><a href="<c:url value="/text?uri=${entry.key}" />">${entry.key}</a></li>
+	<c:forEach items="${resource}" var="entry" varStatus="vs">
+		<c:if test="${empty entry.updatedOn}">
+			<li id="part${vs.index}" class="list-group-item">
+			  <i class="fa fa-spinner fa-pulse"></i> 
+			  Please wait while we are loading the text for you.
+			</li>
+			<script>
+				setTimeout(reload('${entry.uri}','${vs.index}'), 3000);
+				function reload(uri, index)
+				{
+				    $.ajax({
+					  url: "<c:url value="/perspective/researcher/resource"/>",
+					  type : 'GET',
+					  data: { 'uri' : uri },
+					  success: function (rsrc) {
+					    setTimeout(function(){
+					      if(rsrc.updatedOn != null){
+					        var updateResults = "<li class = 'list-group-item'>" +
+					          "<a href = '<c:url value = '/text?uri = " + rsrc.uri + "'/>'>"
```

Comment:
this cannot work. did you test that? c:url is a jsp expression evaluated server-side. at this point you do some javascript which is client-side.